### PR TITLE
Minor fixes in the host and HD-DMA driver

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: src/audio/host.c [hunk 1]
```diff
@@ -359,13 +359,17 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 		/* limit bytes per copy to one period for the whole pipeline
 		 * in order to avoid high load spike
 		 */
-		const uint32_t free_bytes =
-			audio_stream_get_free_bytes(&hd->local_buffer->stream);
```
Comment (plbossart) [inline src/audio/host.c:363]: oh, this declaration shadowed the variable at the higher level. This was completely broken. ARGHHH.

File: src/drivers/intel/hda/hda-dma.c [hunk 2]
```diff
@@ -860,9 +860,13 @@ static int hda_dma_link_check_xrun(struct dma_chan_data *chan)
 	if (chan->direction == DMA_DIR_MEM_TO_DEV && dgcs & DGCS_BUR) {
 		tr_err(&hdma_tr, "hda_dma_link_check_xrun(): underrun detected");
 		dma_chan_reg_update_bits(chan, DGCS, DGCS_BUR, DGCS_BUR);
-	} else if (chan->direction == DMA_DIR_DEV_TO_MEM && dgcs & DGCS_BOR) {
+		return -ENODATA;
+	}
+
+	if (chan->direction == DMA_DIR_DEV_TO_MEM && dgcs & DGCS_BOR) {
```
Comment (plbossart) [inline src/drivers/intel/hda/hda-dma.c:866]: nit-pick: I would add parentheses here and line 860, this sort of code is awful.

**Review by dbaluta — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by ranj063** [comment]
@lgirdwood please do not merge this. It is doing the right thing but this will make every single device fail the pause/release test.

I don't know what the right thing is. 1. continue with green failure while we identify the cause of the xrun reported by the hw or 2. merge this and identify the root cause for the xrun but this will mean all PR tests will fail

**Comment by lgirdwood** [comment]
@ranj063 best to split into separate PRs as it's likely just the one PR thats causing the issue.

**Comment by plbossart** [comment]
@ranj063 the main issue seems to be an ipc timeout?

all the tests with HDaudio in https://sof-ci.01.org/sofpr/PR4797/build10457/devicetest/ give the same pattern:

````
[  429.556672] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x60070000: GLB_STREAM_MSG: TRIG_RELEASE
[  429.803074] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  429.803144] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  430.063097] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x60070000 (msg/reply size: 12/12)
````

File: src/audio/host.c [hunk 3]
```diff
@@ -374,17 +378,6 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
...
-			comp_info(dev, "no bytes to copy, %d free in buffer, %d available in DMA",
-				  audio_stream_get_free_bytes(&hd->local_buffer->stream),
-				  avail_bytes);
-		else
-			comp_info(dev, "no bytes to copy, %d avail in buffer, %d free in DMA",
-				  audio_stream_get_avail_bytes(&hd->local_buffer->stream),
-				  free_bytes);
-	}
-
 	return copy_bytes;
```
Comment (lyakh) [inline src/audio/host.c:381]: you can now just `return ALIGN_DOWN(copy_bytes, hd->dma_copy_align);`

**Review by lyakh — CHANGES_REQUESTED** [review]

**Comment by ranj063** [comment]
> @ranj063 the main issue seems to be an ipc timeout

No @plbossart , the IPC timeout is the result of the xrun. We recently changed the trigger IPC response to be sent from the pipeline task and that's what causes the timeout. I have suggested a fix in #4802

**Comment by plbossart** [comment]
> > @ranj063 the main issue seems to be an ipc timeout
> 
> No @plbossart , the IPC timeout is the result of the xrun. We recently changed the trigger IPC response to be sent from the pipeline task and that's what causes the timeout. I have suggested a fix in #4802

well then let's first deal with the timeout before merging this, otherwise we don't know what we are testing. And if the changes mentioned are already part of the 1.9 release, we'll have to release 1.9.1...

**Comment by lgirdwood** [comment]
> well then let's first deal with the timeout before merging this, otherwise we don't know what we are testing. And if the changes mentioned are already part of the 1.9 release, we'll have to release 1.9.1...

1.9.1 is planned already for fixes for stream offset DMA flows (after they have been validated).

**Comment by ranj063** [comment]
Im going to split this one into 2 PR for host and HD_DMA. closing this one for now

**Pull request closed without merging**